web/src/features/Dashboard/api/getNextAppVersion.ts

```diff
@@ -30,10 +31,11 @@ export const getNextAppVersion = async (appSlug: string) => {
   }
 };
 
-export const useNextAppVersion = (appSlug: string) => {
+export const useNextAppVersion = () => {
+  const { selectedApp } = useSelectedApp();
   return useQuery(
-    ["getNextAppVersion", appSlug],
-    () => getNextAppVersion(appSlug),
+    ["getNextAppVersion", selectedApp?.slug],
+    () => getNextAppVersion(selectedApp?.slug || ""),
```

Comment:
simplifying the interface by removing the need to pass the `appSlug` in. This call is only ever made on the `selectedApp` so we can pull that directly with a hook rather than forcing the components that use this hook to pass the selectedApp slug

Reply:
oh this is nice